test/functional/lua/diagnostic_spec.lua

```diff
@@ -978,6 +978,29 @@ describe('vim.diagnostic', function()
       ]]
       )
     end)
+
+    it('works on blank line #28397', function()
```

Comment:
I tried this locally, the test passes even if I revert the changes to `runtime/lua/vim/diagnostic.lua`

Reply:
fixed